### PR TITLE
Added in functionality for admin.analytics.getFile method

### PR DESCRIPTION
**Pull request by hello-ashleyintech**
###  Summary

Adding in `admin.analytics.getFile` support as mentioned in: https://github.com/slackapi/node-slack-sdk/issues/1191

Link to docs for the method is here: https://api.slack.com/methods/admin.analytics.getFile. This method is a little more unique to other methods -
* In the event of a successful response, this method returns a binary string for a compressed JSON.gz file. Unlike other methods, there is no `ok` property to note its success. Due to the successful response being binary, it needs to be sent as an ArrayBuffer response to be able to be parsed properly - this was also added in this PR
* In the event of a failed response, this method has a similar JSON structure as errors for other methods: `{ ok: false, error: "error message" } `

Tested this functionality + tests locally on my own enterprise grid account.

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: packages/web-api/src/WebClient.ts
Comment (hello-ashleyintech) [inline packages/web-api/src/WebClient.ts:558]: Does this need to be a different type since we're pushing objects from `JSON.parse()` onto it? I tested accessing the output of this and I'm still able to access the objects normally (they aren't converted to strings and are successfully pushed into this array) but I'm wondering if there's a better convention for this?

I saw this convention as highlighted here (https://stackoverflow.com/questions/53964694/how-do-you-declare-an-array-of-objects-inside-typescript) for how to declare an array of objects, but it seems like it would be very involved and complex to declare this interface since we're expecting a large object for each line we're pushing into this array and also since I'm not sure if the output of each line of data from this API call is all expected to be the same structure of object. Would love to get thoughts on this!

File: prod-server-integration-tests/test/admin-web-api-analytics.js
Comment (hello-ashleyintech) [inline prod-server-integration-tests/test/admin-web-api-analytics.js:19]: these dates are hard-coded for the account i used for testing - should I add a note here about how these dates may need to be adjusted for the tests to be able to pass for others' accounts?

File: packages/web-api/src/WebClient.ts
Comment (seratch) [reply packages/web-api/src/WebClient.ts:555]: You can place an interface for the each row data under [the `response` directory](https://github.com/slackapi/node-slack-sdk/tree/main/packages/web-api/src/response). Perhaps, the name can be `AdminAnalyticsGetFileResponseDetails` or so. Also, the auto-generated [`AdminAnalyticsGetFileResponse.ts`](https://github.com/slackapi/node-slack-sdk/blob/%40slack/web-api%406.7.2/packages/web-api/src/response/AdminAnalyticsGetFileResponse.ts) does not have the file data property. Let's skip generating the interface from now on and modify it to have the details property. To prevent future modification by the code generator script, can you add if statement to check if the filename contains admin.analytics.getFile and skip the file generation for it here? https://github.com/slackapi/node-slack-sdk/blob/%40slack/web-api%406.7.2/scripts/code_generator.rb#L56

As for the interface details, if we don't have the bandwidth to work on the details right now, it's okay to go with any type as the first step. But if you want to complete the work this time, you can check the following resources:
* https://api.slack.com/methods/admin.analytics.getFile
* https://github.com/slackapi/java-slack-sdk/blob/v1.24.0/slack-api-client/src/main/java/com/slack/api/methods/response/admin/analytics/AdminAnalyticsGetFileResponse.java#L132-L476

The API document page for this endpoint is well-written. Also, you can refer to the Java SDK code to learn what properties are optional. This endpoint offers three response data patterns (1: public_channel, 2: member, 3: member+metadata_only). So, I think that there are three approaches to define the types:
1. Use unified interface covering all the patterns (most of the properties can be optional)
2. Define two types for the types (AdminAnalyticsGetFileResponseMemberDetails, AdminAnalyticsGetFileResponsePublicChannelDetails) and use a union type for the array
3. Define three types for covering all the possible patterns (~MemberDetails, ~PublicChannelMetadataOnlyDetails, ~PublicChannelDetails) and use a union type for the array

My suggestion here is either Option2 or Option3. Option2 should be great enough perhaps, but if we go with Option3, that'd be perfect for all use cases. Lastly, I know the suggested type names here are very long. I am fine to rename them if you would like to do so.

File: prod-server-integration-tests/test/admin-web-api-analytics.js
Comment (seratch) [reply prod-server-integration-tests/test/admin-web-api-analytics.js:19]: Yeah, it should be great for mitigating future maintainers' confusion!

**Review by seratch — COMMENTED** [review]
Great work!

File: packages/web-api/src/WebClient.ts
Comment (hello-ashleyintech) [reply packages/web-api/src/WebClient.ts:265]: ahh yes, this is a good callout! 🙌  this is actually what i had originally, but I realized that when this API call fails, it returns a regular JSON response, not a GZIP (GZIP only returns on success), so the current logic should cover that use case!

File: packages/web-api/src/WebClient.ts
Comment (hello-ashleyintech) [reply packages/web-api/src/WebClient.ts:554]: this is also a great callout! i was originally using the async method `zlib.unzip()` but was running into issues with accessing the file data since it outputs that into a callback and the data couldn't be reached outside the callback 😢 it seemed complex to have all of the remaining logic that would use this file data within the callback. with that being said, if there's another workaround I could use here, i'm definitely open to it! i'll take a stab at converting this to an async function and see what happens 👀

File: prod-server-integration-tests/test/admin-web-api-analytics.js
Comment (hello-ashleyintech) [reply prod-server-integration-tests/test/admin-web-api-analytics.js:19]: my hesitation with auto-generating the dates is that it may still confuse maintainers since the dates for successful tests depend largely on the enterprise grid account they're using to test and how long it's been active (ex: I created an enterprise grid account 2022-07-05, tried to run this on that day, and got an error that the file wasn't yet available since the first bit of data is available the day after the enterprise grid account is created) 🤔 a good middle ground here might be to add some comments making sure they're using dates that work with the life cycle of the account they're using to test and also noting nuances like the 24 hour period for data being available - would love to get thoughts on this!

File: packages/web-api/src/WebClient.ts
Comment (filmaj) [reply packages/web-api/src/WebClient.ts:265]: Ah great, thanks for confirming!

Comment (hello-ashleyintech) [reply packages/web-api/src/WebClient.ts:270]: update: I did end up adding in some extra logic here similar to what you suggested to account for zlib errors / errors that were thrown when parsing a GZIP!

Comment (hello-ashleyintech) [reply packages/web-api/src/WebClient.ts:270]: a follow up thought: if we can expect every error message in the node SDK to indicate an error by having the response have an `ok: false` property, we could honestly drop the `if / else if` structure here and just use `if ('ok' in result && result.ok === false)` as the conditional

File: prod-server-integration-tests/test/admin-web-api-analytics.js
Comment (hello-ashleyintech) [reply prod-server-integration-tests/test/admin-web-api-analytics.js:22]: added in the comments for the tests  - let me know your thoughts on them and if they're explicit enough!

File: packages/web-api/src/WebClient.ts
Comment (filmaj) [reply packages/web-api/src/WebClient.ts:270]: That's a good point! I am not familiar enough with the API responses to be able to determine if that applies to all APIs. At a minimum, though, would be worthwhile to add this thought as a potential comment / TODO right next to this code. Think that's a worthwhile area of investigation / improvement that we could pursue later.

File: scripts/code_generator.rb
Comment (hello-ashleyintech) [inline scripts/code_generator.rb:60]: @seratch let me know if this location for the `if` statement is what you had in mind to skip the `AdminAnalyticsGetFile` generation! I tested the script and it skipped it, but it also didn't add it into the `response/index.ts` file as it was rewriting it - is this expected behavior? also, since I'm importing the interfaces into `packages/web-api/src/response/index.ts` manually to access them in `WebClient.ts` (since `WebClient` cannot import from the `AdminAnalyticsGetFileResponse` file directly), is this ok since this file can be regenerated by running the script and will override those imports?

File: prod-server-integration-tests/test/admin-web-api-analytics.js
Comment (filmaj) [reply prod-server-integration-tests/test/admin-web-api-analytics.js:22]: Fair enough! That's fine 😄

File: packages/web-api/src/response/AdminAnalyticsGetFileResponse.ts
Comment (filmaj) [inline packages/web-api/src/response/AdminAnalyticsGetFileResponse.ts:13]: Nice!

**Review by filmaj — APPROVED** [review]
Awesome work, looks great to me! Well done

File: packages/web-api/src/response/AdminAnalyticsGetFileResponse.ts
Comment (seratch) [inline packages/web-api/src/response/AdminAnalyticsGetFileResponse.ts:26]: Can you make the indentation consistent (meaning two space indents)?

File: packages/web-api/src/response/AdminAnalyticsGetFileResponse.ts
Comment (seratch) [inline packages/web-api/src/response/AdminAnalyticsGetFileResponse.ts:25]: What do you think naming like `AdminAnalyticsMemberDetails` or so? `MemberDetails` cannot be exposed to external as it's too general.

File: packages/web-api/src/response/index.ts
Comment (seratch) [inline packages/web-api/src/response/index.ts:1]: TODO: This line needs to be manually added in the code generator script (the Ruby script)

File: scripts/code_generator.rb
Comment (seratch) [reply scripts/code_generator.rb:60]: @hello-ashleyintech Good point. The index.ts file needs to be updated correctly. As I mentioned above, you need to add the manually crafted line for AdminAnalyticsGetFileResponse in this script. You can add a new line to do that inside this if statement.

**Review by seratch — CHANGES_REQUESTED** [review]
Thanks for the updates. Great progress

File: scripts/code_generator.rb
Comment (seratch) [inline scripts/code_generator.rb:56]: How about renaming the first arg and expecting an array? The `classes` needs to be an array in the following suggestion:
```suggestion
  def append_multiple_classes_to_index(classes, root_file_name, index_file)
    File.open(index_file, 'a') do |index_f|
      index_f.puts("export { #{classes.join(',')} } from './#{root_file_name}';")
    end
  end
```

File: scripts/code_generator.rb
Comment (seratch) [inline scripts/code_generator.rb:66]: Can you make the change similar by early returning? 

```ruby
    if filename.include? "admin.analytics.getFile"
      classes_to_export = [
        "AdminAnalyticsGetFileResponse",
        "AdminAnalyticsMemberDetails",
        "AdminAnalyticsPublicChannelDetails",
        "AdminAnalyticsPublicChannelMetadataDetails",
        "AdminAnalyticsGetFileResponse",
      ]
      ts_writer.append_multiple_classes_to_index(classes_to_export, index_file)
      return
    end
```

**Review by seratch — CHANGES_REQUESTED** [review]
Great! Two last things..

**Review by seratch — APPROVED** [review]
LGTM! Once the CI builds pass, you can merge this PR (please don't forget squashing the commits into one when clicking the merge button)

**Comment by seratch** [comment]
Okay, this PR is ready to merge. Thanks for adding this feature!

**Pull request merged**
